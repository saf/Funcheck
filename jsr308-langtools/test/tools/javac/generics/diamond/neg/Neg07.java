/*
 * @test /nodynamiccopyright/
 * @bug 6939620
 *
 * @summary  Switch to 'complex' diamond inference scheme
 * @author mcimadamore
 * @compile/fail/ref=Neg07.out Neg07.java -XDrawDiagnostics
 *
 */

class Neg07 {
   static class SuperFoo<X> {}
   static class Foo<X extends Number> extends SuperFoo<X> {
       Foo(X x) {}
   }

   SuperFoo<String> sf1 = new Foo<>("");
   SuperFoo<String> sf2 = new Foo<>("") {};
}
