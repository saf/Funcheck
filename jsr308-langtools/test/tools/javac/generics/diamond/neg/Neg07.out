Neg07.java:17:27: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg07.Foo), (compiler.misc.inferred.do.not.conform.to.bounds: java.lang.String, java.lang.Number)
Neg07.java:18:27: compiler.err.cant.apply.diamond.1: (compiler.misc.diamond: Neg07.Foo), (compiler.misc.inferred.do.not.conform.to.bounds: java.lang.String, java.lang.Number)
2 errors
